bluetooth: ots: Make OTS Object Reads use POSIX parameter types

ekohandel opened the pull request:
It is desirable to have the read and write like functions follow the
POSIX read and write parameter types.

Signed-off-by: Abe Kohandel <abe.kohandel@gmail.com>

ekohandel commented:
The read operation is not expected to return any errors so I used the `size_t` type instead of the `ssize_t` type and consolidated both `obj_read` and `bt_ots_dir_list_content_get` to have an unsigned return type as neither is expected to return an error.

Thalley commented on include/bluetooth/services/ots.h:
```diff
@@ -545,9 +545,9 @@ struct bt_ots_cb {
-	uint32_t (*obj_read)(struct bt_ots *ots, struct bt_conn *conn,
-			     uint64_t id, uint8_t **data, uint32_t len,
-			     uint32_t offset);
+	size_t (*obj_read)(struct bt_ots *ots, struct bt_conn *conn,
+			   uint64_t id, void **data, uint32_t len,
+			   uint32_t offset);
```
Hmm, I see that the previous version simply had a `uint32_t` return value, which implies that the application cannot send any error condition back, and why `size_t` makes sense. 

To be POSIX compliant(ish), I guess we need to change this to a `ssize_t` so that the application may return with an error condition as well (that needs to be checked in `ots` when calling this callback).

ekohandel replied:
I will make a change to the read mechanism to check for errors and terminate the channel if an error condition is returned. This will make it analogous to the write path where an error other than `EINPROGRESS` terminates the channel.

jhedberg replied:
It seems a bit inconsistent to me to have a `uint32_t len` input parameter but `size_t` return value, when those are supposed to be of the same "unit", i.e. the return value tells you how many bytes between `0` and `len` was read.

ekohandel replied:
I have changed the return type to `ssize_t` and the parameters to `size_t` and `off_t`.

carlescufi commented:
@ekohandel could you please follow-up?

ekohandel committed:
bluetooth: ots: Make OTS Object Reads use POSIX parameter types

It is desirable to have the read and write like functions follow the
POSIX read and write parameter types.

Signed-off-by: Abe Kohandel <abe.kohandel@gmail.com>
